### ISSUE TITLE
vmware-vmx Remote Cleanup Fails

### ISSUE THREAD

**Issue by JohnnyBopper** [hashicorp/packer#7404]
Packer Version: 1.3.5
Host Platform: ESXI
Log Output and Config: https://gist.github.com/JohnnyBopper/52420325919c4fd862338a81accb48b7

The build runs fine until it goes to destroy the VM in which case it never succeeds or fails out the build, it gets stuck attempting to destroy the VM.

I am seeing this error in the log:

> Remote command exited with '1': test ! -e "/vmfs/volumes/Packer-Datastore/packer-Win-2016-1552493650-1552493650"

Canceling the build doesn't work and the only way to cancle is to close out the Powershell window. You can see that I tried several times with the following entry in the log:

> Received interrupt signal (count: 23). Ignoring.

Thanks!

**Comment by SwampDragons** [comment]
Thanks for opening. This is at least two bugs -- the actual deletion error, and the fact that the checking loop doesn't time out.

The lack of context cancellation is because it's happening in the cleanup phase but we should definitely have a timeout in here so this doesn't loop infinitely: 
https://github.com/hashicorp/packer/blob/master/builder/vmware/common/step_register.go#L63-L78

**Comment by SwampDragons** [comment]
As for the deletion error... Are you able to manually verify the existance of the /vmfs/volumes/Packer-Datastore/packer-Win-2016-1552493650-1552493650 and is there anything inside of it at the time the builder is stuck spinning?

**Comment by JohnnyBopper** [comment]
After upgrading to 1.4.0 Packer started crashing so I used the version that has the StepConfigureVNC fix (https://github.com/hashicorp/packer/issues/7505). With this version, the destroy step now works. I'm thinking that the issue described in https://github.com/hashicorp/packer/issues/7140 was the same that I was experiencing in regards to the destroy now working.

As far as the cancellation issue, I only saw this happening when the script was stuck in the infinite loop of attempting to destroy the machine. The cancellation interrupt works  during any other step of the build.
